File: test-plugins/org.yakindu.sct.simulation.core.sexec.test/test-gen/org/yakindu/sct/simulation/core/sexec/test/HistoryWithoutInitialStepTest.java
/**
 * Copyright (c) 2015 committers of YAKINDU and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     committers of YAKINDU - initial API and implementation
 */
package org.yakindu.sct.simulation.core.sexec.test;
import org.eclipse.xtext.junit4.InjectWith;
import org.eclipse.xtext.junit4.XtextRunner;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.yakindu.sct.model.sexec.ExecutionFlow;
import org.yakindu.sct.model.sexec.interpreter.test.util.AbstractExecutionFlowTest;
import org.yakindu.sct.model.sexec.interpreter.test.util.SExecInjectionProvider;
import org.yakindu.sct.test.models.SCTUnitTestModels;
import com.google.inject.Inject;
import static org.junit.Assert.assertTrue;
/**
 *  Unit TestCase for HistoryWithoutInitialStep
 */
@SuppressWarnings("all")
@RunWith(XtextRunner.class)
@InjectWith(SExecInjectionProvider.class)
public class HistoryWithoutInitialStepTest extends AbstractExecutionFlowTest {
	@Before
	public void setup() throws Exception {
		ExecutionFlow flow = models
				.loadExecutionFlowFromResource("HistoryWithoutInitialStep.sct");
		initInterpreter(flow);
	}
	@Test
	public void enterThroughInitialEntry() throws Exception {
		interpreter.enter();
		assertTrue(isStateActive("A"));
		raiseEvent("toB");
		interpreter.runCycle();
		assertTrue(isStateActive("C"));
		raiseEvent("next");
		interpreter.runCycle();
		assertTrue(isStateActive("D"));
	}
	@Test
	public void enterCThroughHistory() throws Exception {
		interpreter.enter();
		assertTrue(isStateActive("A"));
		raiseEvent("toB");
		interpreter.runCycle();
		assertTrue(isStateActive("C"));
		raiseEvent("toA");
		interpreter.runCycle();
		assertTrue(isStateActive("A"));
		raiseEvent("toHistory");
		interpreter.runCycle();
		assertTrue(isStateActive("C"));
	}
	@Test
	public void enterDThroughHistory() throws Exception {
		interpreter.enter();
		assertTrue(isStateActive("A"));
		raiseEvent("toB");
		interpreter.runCycle();
		raiseEvent("next");
		interpreter.runCycle();
		assertTrue(isStateActive("D"));
		raiseEvent("toA");
		interpreter.runCycle();
		assertTrue(isStateActive("A"));
		raiseEvent("toHistory");
		interpreter.runCycle();
		assertTrue(isStateActive("D"));
	}
}

File: test-plugins/org.yakindu.sct.generator.c.test/gtests/HistoryWithoutInitialStep/HistoryWithoutInitialStepTest.cc
/**
* Copyright (c) 2017 committers of YAKINDU and others.
* All rights reserved. This program and the accompanying materials
* are made available under the terms of the Eclipse Public License v1.0
* which accompanies this distribution, and is available at
* http://www.eclipse.org/legal/epl-v10.html
*
* Contributors:
*     committers of YAKINDU - initial API and implementation
*/
#include "gtest/gtest.h"
#include "HistoryWithoutInitialStep.h"

static HistoryWithoutInitialStep statechart;

class StatemachineTest : public ::testing::Test{
	protected:
	virtual void SetUp() {
		historyWithoutInitialStep_init(&statechart);
	}
};

void init(){
	historyWithoutInitialStep_enter(&statechart);
	EXPECT_TRUE(historyWithoutInitialStep_isStateActive(&statechart, HistoryWithoutInitialStep_main_region_A));
	historyWithoutInitialStepIface_raise_toB(&statechart);
	historyWithoutInitialStep_runCycle(&statechart);
}

TEST_F(StatemachineTest, enterThroughInitialEntry) {					
	init();
	EXPECT_TRUE(historyWithoutInitialStep_isStateActive(&statechart, HistoryWithoutInitialStep_main_region_B_r1_C));
	historyWithoutInitialStepIface_raise_next(&statechart);
	historyWithoutInitialStep_runCycle(&statechart);
	EXPECT_TRUE(historyWithoutInitialStep_isStateActive(&statechart, HistoryWithoutInitialStep_main_region_B_r1_D));
}
TEST_F(StatemachineTest, enterCThroughHistory) {					
	init();
	EXPECT_TRUE(historyWithoutInitialStep_isStateActive(&statechart, HistoryWithoutInitialStep_main_region_B_r1_C));
	historyWithoutInitialStepIface_raise_toA(&statechart);
	historyWithoutInitialStep_runCycle(&statechart);
	EXPECT_TRUE(historyWithoutInitialStep_isStateActive(&statechart, HistoryWithoutInitialStep_main_region_A));
	historyWithoutInitialStepIface_raise_toHistory(&statechart);
	historyWithoutInitialStep_runCycle(&statechart);
	EXPECT_TRUE(historyWithoutInitialStep_isStateActive(&statechart, HistoryWithoutInitialStep_main_region_B_r1_C));
}
TEST_F(StatemachineTest, enterDThroughHistory) {					
	init();
	historyWithoutInitialStepIface_raise_next(&statechart);
	historyWithoutInitialStep_runCycle(&statechart);
	EXPECT_TRUE(historyWithoutInitialStep_isStateActive(&statechart, HistoryWithoutInitialStep_main_region_B_r1_D));
	historyWithoutInitialStepIface_raise_toA(&statechart);
	historyWithoutInitialStep_runCycle(&statechart);
	EXPECT_TRUE(historyWithoutInitialStep_isStateActive(&statechart, HistoryWithoutInitialStep_main_region_A));
	historyWithoutInitialStepIface_raise_toHistory(&statechart);
	historyWithoutInitialStep_runCycle(&statechart);
	EXPECT_TRUE(historyWithoutInitialStep_isStateActive(&statechart, HistoryWithoutInitialStep_main_region_B_r1_D));
}


